knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -111,13 +111,14 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 
 // configure default settings for OpenShift
 func (r *ReconcileKnativeServing) configure(instance *servingv1alpha1.KnativeServing) error {
-	if _, ok := instance.GetAnnotations()[common.MutationTimestampKey]; ok {
-		return nil
-	}
 	log.Info("Configuring KnativeServing for OpenShift")
-	if err := common.Mutate(instance, r.client); err != nil {
+	changed, err := common.Mutate(instance, r.client)
```

Comment:
I understand why we need to overwrite the image overrides, i.e. make `Configure` less conservative, but I don't understand why we need to know we changed something here.

This has also made me realize that we're not setting `version/version.go` properly. The operator should know definitively what version it's installing, so it should be able to tell exactly which images it should be configuring.

Reply:
I want to know if something changed here to not unnecessarily make the api call to update the KnativeServing.

The second part I'm not quite parsing. We're configuring the correct images via env I think?